### PR TITLE
♻️ Rewrite Dialog stories in Typescript

### PR DESCRIPTION
**Pull request by vnys**
Part of #735

### CODE REVIEW

File: apps/storybook-react/stories/Dialog.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    </Fragment>
+  ),
+}
+
+const ACTION_CHOICES = {
+  none: null,
+  buttons: (
+    <TempButtonWrapper>
+      <Button>OK</Button>
+      <Button variant="ghost">Cancel</Button>
```
Comment (vnys) [inline apps/storybook-react/stories/Dialog.stories.tsx:91]: Ok left, cancel right…

File: apps/storybook-react/stories/Dialog.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+export const Default: Story<DialogProps> = () => (
+  <Dialog>
+    <Title>{TITLE_CHOICES['text']}</Title>
+    <CustomContent>{CUSTOM_CONTENT_CHOICES['description']}</CustomContent>
+    <Actions>
+      <TempButtonWrapper>
+        <Button>Cancel</Button>
+        <Button>OK</Button>
```
Comment (vnys) [inline apps/storybook-react/stories/Dialog.stories.tsx:103]: Cancel left, OK right… 🤔 I think this has gone unnoticed in the original story…

**Comment by vnys** [comment]
Discovered something in the story that I removed since it’s different from what we do elsewhere. If we _don’t_ export the type explicitly, then we can still do `type DialogProps = typeof Dialog` in the story just like we do in the index.ts exports 🤷

File: apps/storybook-react/stories/Dialog.stories.tsx (hunk 2 above)
Comment (vnys) [reply apps/storybook-react/stories/Dialog.stories.tsx:103]: image

Comment (vnys) [reply apps/storybook-react/stories/Dialog.stories.tsx:103]: So affirmative actions always on the left, destructive actions on the right.

Comment (mimarz) [reply apps/storybook-react/stories/Dialog.stories.tsx:102]: Is it not the other way around? 🙈

Comment (vnys) [reply apps/storybook-react/stories/Dialog.stories.tsx:102]: This is from Figma, but I don’t know _who_ made that in Figma. Maybe there’s something in the Storefront…

Comment (vnys) [reply apps/storybook-react/stories/Dialog.stories.tsx:102]: Can confirm that this is correct

**Review by pomfrida — APPROVED** [review]
LGTM 👍

File: libraries/core-react/src/Dialog/index.ts [hunk 3]
```diff
@@ -3,16 +3,16 @@ import { Actions } from './Actions'
...
 
-const Dialog = BaseComponent as DialogTypes
+const Dialog = BaseComponent as DialogProps
 
 Dialog.Actions = Actions
 Dialog.Title = Title
 Dialog.CustomContent = CustomContent
 
-export { Dialog }
+export { Dialog, DialogProps }
```
Comment (mimarz) [inline libraries/core-react/src/Dialog/index.ts:18]: Didn't we agree to export the actual `DialogProps` here? @wenche @vnys

**Review by mimarz — CHANGES_REQUESTED** [review]
Looks ok, but this it does not follow the same pattern we have for our other compound components types...

File: libraries/core-react/src/Dialog/index.ts (hunk 3 above)
Comment (vnys) [reply libraries/core-react/src/Dialog/index.ts]: Yes we did

**Comment by vnys** [comment]
> this it does not follow the same pattern we have for our other compound components types

That was not intentional, it should now

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**